### ISSUE TITLE
TorchNet support of Mask-R-CNN

### ISSUE THREAD

**Issue by hhbyyh** [intel-analytics/analytics-zoo#1620]
Use this issue to track the progress on supporting https://github.com/facebookresearch/maskrcnn-benchmark with TorchNet.

**Comment by hhbyyh** [comment]
test code: 
```
from maskrcnn_benchmark.config import cfg
from demo.predictor import COCODemo
from demo.predictor import to_image_list
import cv2
import torch
from torch import nn
from maskrcnn_benchmark.structures.image_list import ImageList
from maskrcnn_benchmark.structures.image_list import to_image_list

config_file = "../configs/caffe2/e2e_mask_rcnn_R_50_FPN_1x_caffe2.yaml"

# update the config options with the config file
cfg.merge_from_file(config_file)
# manual override some options
cfg.merge_from_list(["MODEL.DEVICE", "cpu"])

coco_demo = COCODemo(
    cfg,
    min_image_size=800,
    confidence_threshold=0.7,
)

import glob
import cv2

class MaskRCNNWrapper(nn.Module):
    def __init__(self):
        super(MaskRCNNWrapper, self).__init__()
        self.core = coco_demo.model

    def forward(self, img):
        imageList = to_image_list(img)

        x = self.core(imageList)[0]
        bbox = x.bbox
        labels = x.extra_fields['labels']
        mask = x.extra_fields['mask']
        scores = x.extra_fields['scores']
        return bbox, labels, mask, scores

# load image and then run prediction
original_image = cv2.imread("/home/yuhao/workspace/github/facebookresearch/maskrcnn-benchmark/demo/cats/cat.72.jpg")
# predictions = coco_demo.run_on_opencv_image(image)
# print(predictions)

image = coco_demo.transforms(original_image)
# convert to an ImageList, padded so that it is divisible by
# cfg.DATALOADER.SIZE_DIVISIBILITY
image_list = to_image_list(image, coco_demo.cfg.DATALOADER.SIZE_DIVISIBILITY)
image_list = image_list.to(coco_demo.device)

wrapper = MaskRCNNWrapper()

# compute predictions
with torch.no_grad():
    predictions = wrapper(image_list.tensors)


print(predictions)


from zoo.common.nncontext import *
from zoo.pipeline.nnframes import *
from zoo.pipeline.api.net.torch_net import TorchNet
from pyspark.sql import SparkSession
from pyspark.sql.types import DoubleType
from zoo.pipeline.api.net.torch_criterion import TorchCriterion

sparkConf = init_spark_conf().setAppName("test_pytorch_lenet").setMaster("local[1]") \
    .set('spark.driver.memory', '10g')
sc = init_nncontext(sparkConf)
spark = SparkSession.builder.config(conf=sparkConf).getOrCreate()

net = TorchNet.from_pytorch(wrapper, [1, 3, 800, 1184])

```

**Comment by hhbyyh** [comment]
The maskrcnn repo has issues and PRs about the jit tracing support:

https://github.com/facebookresearch/maskrcnn-benchmark/pull/138
https://github.com/facebookresearch/maskrcnn-benchmark/issues/27

Right now it's still working in progress.

**Comment by wzhongyuan** [comment]
Saw this one https://pytorch.org/docs/stable/jit.html#disable-jit-for-debugging

**Comment by hhbyyh** [comment]
Torchvision now has include mask r cnn, yet the JIT support is still not finished.

https://github.com/pytorch/vision/issues/1002.

I'm trying to verify the related PRs manually but met build error.
